### DISCUSSION TITLE
Generate representation string with rich

### DISCUSSION THREAD

**Discussion by eendebakpt** [Textualize/rich#2597] [Q&A]
Using rich I can print nice representations, but how can I generate a string with the equivalent representation?

An example
```
from dataclasses import dataclass
from rich import print as rprint
from rich import pretty

@dataclass
class Test:
        a: str
        b: int
        c: float
        
t=Test('hi', 1, 2.)
rprint(t)         # looks good

t # in the environment (spyder or jupyter) looks plain
```
I would like the output of `t` to be formatted in the style of rich. I can use `pretty.install()`, but that applies to all objects, not just by dataclass `Test`.
Can I do something like
```
@dataclass
class Test:
        a: str
        b: int
        c: float
        
        def _repr_pretty_(self, p, cycle):
            return rich.pretty_format(self)
```
with rich?

**Comment by jacanchaplais** [comment]
I am also trying to achieve this. The `__rich__` special method works nicely, but `rich.pretty.install()` seems to need to be called for my IPython terminal to hook into it. This colourises everything, though, so I feel a bit bad putting this in the `__init__.py` of my package, because my users might not like having their whole experience pretty printed. Not sure what the workaround is.

**Reply by jacanchaplais** [reply]
Docs on `__rich__` special method, btw https://rich.readthedocs.io/en/stable/protocol.html
